File: task03/java/subtask4.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;

public class subtask4 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new FileReader("input.txt"));
        int n = Integer.parseInt(reader.readLine());

        BufferedWriter writer = new BufferedWriter(new FileWriter("output.txt"));

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n - i - 1; j++) {
                writer.write(" ");
            }
            for (int j = 0; j < 2 * i + 1; j++) {
                writer.write("*");
            }
            writer.write("\n");
        }

        for (int i = n - 2; i >= 0; i--) {
            for (int j = 0; j < n - i - 1; j++) {
                writer.write(" ");
            }
            for (int j = 0; j < 2 * i + 1; j++) {
                writer.write("*");
            }
            writer.write("\n");
        }

        writer.close();
    }
}

File: task03/rust/subtask4.rs
use std::fs::File;
use std::io::{self, BufReader, BufWriter, Write};

fn main() -> io::Result<()> {
    let file = File::open("input.txt")?;
    let reader = BufReader::new(file);
    let mut input = String::new();
    reader.read_line(&mut input)?;
    let n: i32 = input.trim().parse().expect("Please type a number");

    let file = File::create("output.txt")?;
    let mut writer = BufWriter::new(file);

    for i in 0..n {
        for _ in 0..n - i - 1 {
            writer.write(b" ")?;
        }
        for _ in 0..2 * i + 1 {
            writer.write(b"*")?;
        }
        writer.write(b"\n")?;
    }

    for i in (0..n - 1).rev() {
        for _ in 0..n - i - 1 {
            writer.write(b" ")?;
        }
        for _ in 0..2 * i + 1 {
            writer.write(b"*")?;
        }
        writer.write(b"\n")?;
    }

    Ok(())
}